src/stepfunctions/steps/states.py

```diff
@@ -218,9 +218,24 @@ def next(self, next_step):
         Returns:
             State or Chain: Next state or chain that will be transitioned to.
         """
-        if self.type in ('Choice', 'Succeed', 'Fail'):
+        if self.type in ('Succeed', 'Fail'):
             raise ValueError('Unexpected State instance `{step}`, State type `{state_type}` does not support method `next`.'.format(step=next_step, state_type=self.type))
 
+        # By design, Choice states do not have the Next field. Their purpose is to define conditional transitions based
```

Comment:
We should document the behaviour of chaining Choice states publicly. Consumers of the SDK won't be looking at the source code. Something along the lines of:

> When used in a chain, the subsequent step becomes the default choice that executes if none of the specified rules match.

https://github.com/aws/aws-step-functions-data-science-sdk-python/blob/main/src/stepfunctions/steps/states.py#L402-L470

The docstrings in our source code become part of the public documentation and are visible in user's IDE: https://aws-step-functions-data-science-sdk.readthedocs.io/en/stable/states.html#stepfunctions.steps.states.Choice

Reply:
Very good point! Will make sure to update the public documentation as well.